api/tests/integration/EntityTest.php

```diff
@@ -58,6 +61,107 @@ public function testGetAll()
         $this->assertJsonMultipleEntries();
     }
 
+    public function testGetAllPaginated()
+    {
+        $defaultLimit = \App\Http\Controllers\ApiController::PAGINATOR_DEFAULT_LIMIT;
+        $entities = factory(App\Models\TestEntity::class, $defaultLimit+1)->create();
+        $this->get('/test/entities/pages');
+        $this->assertResponseOk();
+        $this->shouldReturnJson();
+        $this->assertJsonArray();
+        $this->assertJsonMultipleEntries();
+
+        $object = json_decode($this->response->getContent());
+        $this->assertEquals($defaultLimit, count($object));
+    }
+
+    public function testGetAllPaginatedSimpleRange()
+    {
+        $entities = factory(App\Models\TestEntity::class, 20)->create();
+        $totalCount = $this->repository->count();
+        $this->get('/test/entities/pages', ['Range'=>'0-19']);
+        $object = json_decode($this->response->getContent());
+        $this->assertResponseStatus(206);
+        $this->shouldReturnJson();
+        $this->assertJsonArray();
+        $this->assertJsonMultipleEntries();
+        $this->assertEquals(20, count($object));
+        list($first, $last, $total) = $this->parseRange($this->response->headers->get('content-range'));
+        $this->assertEquals($total, $totalCount);
+        $this->assertEquals($first, 0);
+        $this->assertEquals($last, 19);
+    }
+
+    public function testPaginationBadRanges()
+    {
+        $entities = factory(App\Models\TestEntity::class, 20)->create();
+        $this->get('/test/entities/pages', ['Range'=>'19-18']);
+        $this->assertResponseStatus(416);
+    }
```

Comment:
This should probably be a 400 Bad Request status, as it isn't `416 Requested Range Not Satisfiable` due to the requested range being out of bounds; it is a logic error on the part of the client.